Sources/PrimerSDK/Classes/User Interface/Apple Pay/ApplePayViewModel.swift

```diff
@@ -1,69 +1,317 @@
+            supportedNetworks.append(.carteBancaire)
+        }
+
+        if #available(iOS 12.0, *) {
+//            @available(iOS 12.0, *)
+            supportedNetworks.append(.eftpos)
+            supportedNetworks.append(.electron)
+            supportedNetworks.append(.maestro)
+            supportedNetworks.append(.vPay)
+        }
+
+        if #available(iOS 12.1.1, *) {
+//            @available(iOS 12.1.1, *)
+            supportedNetworks.append(.elo)
+            supportedNetworks.append(.mada)
+        }
+        
+        if #available(iOS 10.3.1, *) {
+//            @available(iOS 10.3, *)
+            supportedNetworks.append(.idCredit)
+        }
+        
+        if #available(iOS 10.1, *) {
+//            @available(iOS 10.1, *)
+            supportedNetworks.append(.JCB)
+            supportedNetworks.append(.suica)
+        }
+        
+        if #available(iOS 10.3, *) {
+//            @available(iOS 10.3, *)
+            supportedNetworks.append(.quicPay)
+        }
+        
+        if #available(iOS 14.0, *) {
+//            @available(iOS 14.0, *)
+//            supportedNetworks.append(.barcode)
+            supportedNetworks.append(.girocard)
+        }
+        
+        if PKPaymentAuthorizationViewController.canMakePayments(usingNetworks: supportedNetworks) {
+            let request = PKPaymentRequest()
+            request.currencyCode = applePayRequest.currency.rawValue
+            request.countryCode = applePayRequest.countryCode.rawValue
+            request.merchantIdentifier = "merchant.primer.dev.evangelos"
+            request.merchantCapabilities = [.capability3DS]
+            request.supportedNetworks = supportedNetworks
+            request.paymentSummaryItems = applePayRequest.items.compactMap({ $0.applePayItem })
+            
+            guard let paymentVC = PKPaymentAuthorizationViewController(paymentRequest: request) else {
+                log(logLevel: .error, title: "APPLE PAY", message: "Unable to present Apple Pay authorization.")
+                return completion(AppleException.unableToPresentApplePay)
             }
+            
+            paymentVC.delegate = self
+            
+            applePayCompletion = { [weak self] result in
+                switch result {
+                case .success(let applePayPaymentResponse):
+                    // Loading view
+                    let settings: PrimerSettingsProtocol = DependencyContainer.resolve()
+                    
+                    let applePayService: ApplePayServiceProtocol = DependencyContainer.resolve()
+                    applePayService.fetchConfig { [weak self] (err) in
+                        if let err = err {
+                            DispatchQueue.main.async {
+                                Primer.shared.presentingViewController?.dismiss(animated: true, completion: {
+                                    if !settings.hasDisabledSuccessScreen {
```

Comment:
[Feedback]
Apple Pay shouldn't care about what's after 🤔 
It could be a success screen or it could be something else in the future. Is it possible to reduce the coupling between Apple Pay and the bottom sheet?

Reply:
Same as here https://github.com/primer-io/primer-sdk-ios/pull/72#discussion_r639692573